pkg/resourcemanager/controller/secret/reconciler.go

```diff
@@ -66,6 +68,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels[v1alpha1.ReferencedBy]; !ok {
```

Comment:
You could update https://github.com/gardener/gardener/blob/b46a19d66fbba1e8bf12387eed69595b6e5a2f91/pkg/resourcemanager/controller/secret/reconciler.go#L35. The reconciler no longer maintains a finalizer only.